File: system-test/dotnet/SystemTests/Legacy/Mod05/SmokeTests/System/ShopBaseSmokeTest.cs
using Common;
using SystemTests.Legacy.Mod05.Base;
using Shouldly;
using Xunit;

namespace SystemTests.Legacy.Mod05.SmokeTests.System;

public abstract class ShopBaseSmokeTest : BaseDriverTest
{
    protected abstract Task SetUpShopDriverAsync();

    public override async Task InitializeAsync()
    {
        await SetUpShopDriverAsync();
    }

    [Fact]
    public async Task ShouldBeAbleToGoToShop()
    {
        var result = await _shopDriver!.GoToShopAsync();
        result.ShouldBeSuccess();
    }
}

File: system-test/java/src/test/java/com/optivem/shop/systemtest/legacy/mod05/smoke/system/ShopBaseSmokeTest.java
package com.optivem.shop.systemtest.legacy.mod05.smoke.system;

import com.optivem.shop.systemtest.legacy.mod05.base.BaseDriverTest;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static com.optivem.shop.dsl.common.ResultAssert.assertThatResult;

public abstract class ShopBaseSmokeTest extends BaseDriverTest {
    @BeforeEach
    void setUp() {
        setShopDriver();
    }

    protected abstract void setShopDriver();

    @Test
    void shouldBeAbleToGoToShop() {
        var result = shopDriver.goToShop();
        assertThatResult(result).isSuccess();
    }
}

